bazel/re2c.bzl

```diff
@@ -0,0 +1,72 @@
+# re2c generator rules - assuming that re2c is available as '@re2c//:re2c'.
+
+load("@bazel_skylib//lib:paths.bzl", "paths")
+
+visibility("public")
+
+# Define supported extensions as defined by:
+# . https://bazel.build/reference/be/c-cpp#cc_library_args
+# . https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/rules/cpp/CppFileTypes.java
+_C_SRC_EXT = [".c", ".cc", ".cpp", ".cxx", ".c++", ".C", ".cu", ".cl"]
+_C_HDR_EXT = [".h", ".hh", ".hpp", ".ipp", ".hxx", ".h++", ".inc", ".inl", ".tlh", ".tli", ".H", ".tcc"]
+
+def _re2c_impl(ctx):
+    srcs = ctx.files.srcs
+    if len(srcs) != 1:
+        fail("Attribute `srcs` must contain exactly 1 '.re' or '.re2c' file.")
+    outs = ctx.outputs.outs
+    if not outs or [o for o in outs if paths.split_extension(o.path)[1] not in _C_SRC_EXT + _C_HDR_EXT]:
+        fail("Attribute `outs` must contain at least 1 '.cc' or '.h' file.")
```

Comment:
Nit: the error message is now outdated as it suggests only `'.cc' or '.h'` are accepted. Maybe say something more generic, like `at least one C/C++ source or header file`?

Reply:
Done.